src/Algebra/Graph/Acyclic/AdjacencyMap.hs

```diff
@@ -0,0 +1,112 @@
+module Algebra.Graph.Acyclic.AdjacencyMap where
+
+import qualified Algebra.Graph.AdjacencyMap as AM
+import qualified Algebra.Graph.AdjacencyMap.Algorithm as AMA
+import qualified Algebra.Graph.AdjacencyMap.Internal as AMI
+import qualified Algebra.Graph.NonEmpty.AdjacencyMap as NonEmpty
+import Control.Monad.Trans.Class (lift)
+import Control.Monad.Trans.State.Strict
+import Data.Map (Map, (!))
+import qualified Data.Map.Strict as Map
+import qualified Data.Set as Set
+
+newtype AdjacencyMap a = AAM
+  { aam :: AM.AdjacencyMap a
+  } deriving (Show)
+
+consistent :: Ord a => AdjacencyMap a -> Bool
+consistent (AAM m) = AMI.consistent m && AMA.isAcyclic m
+
+type AdjacencyMapP a = AdjacencyMap (Int, a)
```

Comment:
Why do we need this?

Reply:
To impose a different order on the vertices this is required.
The graph is always constructed keeping the order of vertices in mind.

Reply:
Which requirement are you trying to achieve using this extra complexity?

Reply:
Adding edges in the near future. But I've changed it now!